dpnp/tests/test_product.py

```diff
@@ -1196,6 +1140,54 @@ def test_large(self, shape1, shape2):
         expected = numpy.matmul(a, b)
         assert_dtype_allclose(result, expected, factor=24)
 
+    @pytest.mark.parametrize("dtype", _selected_dtypes)
+    def test_large_values(self, dtype):
+        val = 94906267
+        a = numpy.array([[[val, val], [val, val]]], dtype=dtype)
+        b = numpy.array([94906265, 94906265], dtype=dtype)
+        ia, ib = dpnp.array(a), dpnp.array(b)
+
+        result = dpnp.matmul(ia, ib)
+        expected = numpy.matmul(a, b)
+        assert_dtype_allclose(result, expected)
+
+    @pytest.mark.parametrize("dt_out", [numpy.int32, numpy.float32])
+    @pytest.mark.parametrize(
+        "shape1, shape2",
+        [((2, 4), (4, 3)), ((4, 2, 3), (4, 3, 5))],
+        ids=["gemm", "gemm_batch"],
+    )
+    def test_special_case(self, dt_out, shape1, shape2):
+        # Although inputs are int, gemm will be used for calculation
+        a = numpy.ones(shape1, dtype=numpy.int8)
+        b = numpy.ones(shape2, dtype=numpy.int8)
```

Comment:
it'd be good to use random data for int8 also:
```suggestion
        a = generate_random_numpy_array(shape1, dtype=numpy.int8)
        b = generate_random_numpy_array(shape2, dtype=numpy.int8)
```